source/server/session_repository.cpp

```diff
@@ -76,46 +71,37 @@ int SessionRepository::add_session(
 // the ID of the session that matches the given ID or name, or 0 if such a session is not found.
 // Only one of the two parameters needs to be specified.
 // Passing only a name returns the corresponding ID.
-uint32_t SessionRepository::access_session(uint32_t session_id, const std::string& session_name)
+std::string SessionRepository::access_session(const std::string& session_name)
 {
   std::unique_lock<std::shared_mutex> lock(repository_lock_);
   auto now = std::chrono::steady_clock::now();
   auto it = named_sessions_.find(session_name);
   if (it != named_sessions_.end()) {
     it->second->last_access_time = now;
-    return it->second->id;
+    return it->second->name;
   }
-  auto sessions_it = sessions_.find(session_id);
-  if (sessions_it != sessions_.end()) {
-    sessions_it->second->last_access_time = now;
-    return session_id;
-  }
-  return 0;
+  return "";
 }
 
 // Removes a session by ID.
 // To remove a session by name, use access_session to get the session ID.
-void SessionRepository::remove_session(uint32_t id)
+void SessionRepository::remove_session(const std::string& name)
 {
   std::unique_lock<std::shared_mutex> lock(repository_lock_);
-  auto it = sessions_.find(id);
-  if (it != sessions_.end()) {
-    auto named_it = named_sessions_.find(it->second->name);
-    if (named_it != named_sessions_.end()) {
-      named_sessions_.erase(named_it);
-    }
-    sessions_.erase(it);
+  auto named_it = named_sessions_.find(name);
+  if (named_it != named_sessions_.end()) {
+    named_sessions_.erase(named_it);
   }
 }
 
-void SessionRepository::register_dependent_session(uint32_t id, uint32_t dependent_session_id, std::function<void()> cleanup)
+void SessionRepository::register_dependent_session(const std::string& name, const std::string& dependent_session_name, std::function<void()> cleanup)
```

Comment:
I can't see the results from `run_win_system_tests` (I haven't spent much time that action, I'm not sure if that's normal). Could you tell if the `rfmx` tests ran and passed? Those are windows-only and those drivers are the main clients of dependent sessions.

Reply:
I'll try and run them locally.  The RFMX tests unit and integration tests which use dependent sessions do pass. I did break something there, so it seems those tests are good.

Reply:
Yeah, those tests seem good for that.

I am finding as I'm looking at other PRs that the `run_win_system_tests` either are failing with no description in the action or are now infinitely queuing so something's messed up there, probably with our system. Not sure yet but looking into it.

Reply:
I kicked off the pipeline to re-create the VM the `run_win_system_tests` connects to. After that completes, I'll re-queue that action on this PR to see if it actually runs this time.

Reply:
It looks like the system tests are still failing.
I ran what tests I could locally.  many of the RF tests fail for what looks like other reasons. I think maybe we need actual HW to run the tests?

Reply:
System tests finally ran and completed successfully.